File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string context);
int count_words(string context);
int count_sentence(string context);
int index_grade(int letters, int sentences, int words);

int main(void)
{
    string context = get_string(" Enter the paragraph =");

    int letters = count_letters(context);
    int words = count_words(context);
    int sentences = count_sentence(context);
    int grade = index_grade(letters, sentences, words);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade == 1)
    {
        printf(" Grade 1\n");
    }
    else if (grade == 2)
    {
        printf(" Grade 2\n");
    }
    else if (grade == 3)
    {
        printf(" Grade 3\n");
    }
    else if (grade == 4)
    {
        printf(" Grade 4\n");
    }
    else if (grade == 5)
    {
        printf(" Grade 5\n");
    }
    else if (grade == 6)
    {
        printf(" Grade 6\n");
    }
    else if (grade == 7)
    {
        printf(" Grade 7\n");
    }
    else if (grade == 8)
    {
        printf(" Grade 8\n");
    }
    else if (grade == 9)
    {
        printf(" Grade 9\n");
    }
    else if (grade == 10)
    {
        printf(" Grade 10\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int count_letters(string context)
{
    int l = 0;
    int len = strlen(context);
    for (int i = 0; i < len; i++)
    {
        if (isalpha(context[i]))
        {
            l++;
        }
    }
    return l;
}

int count_words(string context)
{
    int w = 1;
    int len = strlen(context);
    for (int i = 0; i < len; i++)
    {
        if (isspace(context[i]))
        {
            w++;
        }
    }
    return w;
}

int count_sentence(string context)
{
    int s = 0;
    int len = strlen(context);
    for (int i = 0; i < len; i++)
    {
        if (context[i] == '.' || context[i] == '!' || context[i] == '?')
        {
            s++;
        }
    }
    return s;
}

int index_grade(int letters, int sentences, int words)
{
    float L = (letters / (float) words * 100);
    float S = (sentences / (float) words * 100);

    int index = round(0.0588 * L - 0.296 * S - 15.8);
    return index;
}

File: readability.py
from cs50 import get_string


def count_letters(context):  # counting letters function
    l = 0
    for i in context:
        if i.isalpha():
            l += 1
    return l


def count_words(context):  # counting words function
    w = 1
    for i in context:
        if i.isspace():
            w += 1
    return w


def count_sentences(context):  # counting sentences function
    s = 0
    for i in context:
        if i in ".!?":
            s += 1
    return s


def index_grade(letters, sentences, words):                # Coleman-Liau index grading formula
    L = float(letters / words * 100)
    S = float(sentences/words * 100)
    index = int(round(0.0588 * L - 0.296 * S - 15.8))
    return index


context = get_string(" Enter the paragraph =")  # getting the context

letters = count_letters(context)
words = count_words(context)
sentences = count_sentences(context)

grade = index_grade(letters, sentences, words)

if grade < 1:  # conditions basis on grades
    print("Before Grade 1")

elif grade == 1:
    print("Grade 1")

elif grade == 2:
    print("Grade 2")

elif grade == 3:
    print("Before Grade 3")

elif grade == 4:
    print("Before Grade 4")

elif grade == 5:
    print("Before Grade 5")

elif grade == 6:
    print("Before Grade 6")

elif grade == 7:
    print("Before Grade 7")

elif grade == 8:
    print("Before Grade 8")

elif grade == 9:
    print("Before Grade 9")

elif grade == 10:
    print("Before Grade 10")

else:
    print("Grade 16+")
